temba/triggers/views.py

```diff
@@ -774,6 +808,70 @@ def get_form_kwargs(self):
             kwargs["auto_id"] = "id_schedule_%s"
             return kwargs
 
+    class ScheduleInBatch(CreateTrigger):
+        form_class = ScheduleTriggerInBatchForm
+        title = _("Create Schedule In Batch")
+
+        def get_context_data(self, **kwargs):
+            context = super().get_context_data(**kwargs)
+
+            params_context = flow_params_context(self.request)
+            context.update(params_context)
+
+            return context
+
+        def form_invalid(self, form):
+            if "_format" in self.request.GET and self.request.GET["_format"] == "json":  # pragma: needs cover
+                return HttpResponse(
+                    json.dumps(dict(status="error", errors=form.errors)), content_type="application/json", status=400
+                )
+            else:
+                return super().form_invalid(form)
+
+        def create_trigger(self, start_time, org, form):
+            schedule = Schedule.create_schedule(
+                org,
+                self.request.user,
+                start_time,
+                form.cleaned_data.get("repeat_period"),
+                repeat_days_of_week=form.cleaned_data.get("repeat_days_of_week"),
+            )
+
+            return Trigger.objects.create(
+                flow=self.form.cleaned_data["flow"],
+                org=self.request.user.get_org(),
+                schedule=schedule,
+                trigger_type=Trigger.TYPE_SCHEDULE,
+                created_by=self.request.user,
+                modified_by=self.request.user,
+                extra=self.flow_params,
+            )
+
+        def form_valid(self, form):
+            analytics.track(self.request.user.username, "temba.trigger_created", dict(type="schedule"))
+            org = self.request.user.get_org()
+            start_time = form.cleaned_data["start_datetime"]
+            recipients = self.form.cleaned_data["omnibox"]
+            batch_interval = self.form.cleaned_data["batch_interval"]
+
+            triggers = []
+            for group in recipients["groups"]:
+                start_time = start_time + timedelta(minutes=int(batch_interval))
```

Comment:
I think the first one should be on time that the user selected, and from the second one, we can start adding the batch interval.

Reply:
yup, good catch